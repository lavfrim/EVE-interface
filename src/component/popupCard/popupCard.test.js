import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import PopupCard from './index';

const corporation = {
    name: 'corp name',
}

describe('PopupCard should render correctly', () => {
    it('with stage = basic', () => {
        const wrapper = mount(
            <PopupCard stage={0} corporation={corporation} />
        );
        wrapper.setState({ ceoInfo: {
            name: 'ceo Name',
        } });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('with stage = clicked', () => {
        const wrapper = mount(
            <PopupCard stage={1} corporation={corporation} />
        );
        wrapper.setState({ ceoInfo: {
            name: 'ceo Name',
        } });
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

