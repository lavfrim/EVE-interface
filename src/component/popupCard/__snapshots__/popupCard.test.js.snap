// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupCard should render correctly with stage = basic 1`] = `
<PopupCard
  corporation={
    Object {
      "name": "corp name",
    }
  }
  stage={0}
>
  <CorporationCard
    ceoInfo={
      Object {
        "name": "ceo Name",
      }
    }
    corporation={
      Object {
        "name": "corp name",
      }
    }
  >
    <p>
      Corporation name: corp name
    </p>
    <div
      className="corporation-card__ceo-link"
    >
      <p
        className="corporation-card__ceo-link-description"
      >
        CEO name:
      </p>
      <div
        className="corporation-card__ceo-link-link"
        onClick={[Function]}
      >
        ceo Name
      </div>
    </div>
    <p>
      Member count: undefined
    </p>
    <p>
      Description: undefined
    </p>
  </CorporationCard>
</PopupCard>
`;

exports[`PopupCard should render correctly with stage = clicked 1`] = `
<PopupCard
  corporation={
    Object {
      "name": "corp name",
    }
  }
  stage={1}
>
  <CEOCard
    ceoInfo={
      Object {
        "name": "ceo Name",
      }
    }
  >
    <p>
      CEO name: ceo Name
    </p>
    <p>
      Birthday: Invalid Date
    </p>
    <div
      className="ceo-card__race"
    >
      <p
        className="ceo-card__race-description"
      >
        Race:
      </p>
      <Loading>
        <span>
          Loading...
        </span>
      </Loading>
    </div>
  </CEOCard>
</PopupCard>
`;
